internal/pgengine/actions.go

```diff
@@ -0,0 +1,50 @@
+package pgengine
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+)
+
+// ResetInstance attempts to reset the cluster to a clean state.
+// It deletes all cluster level objects, i.e., roles, which are not deleted
+// by dropping database(s). This can be useful for re-using a cluster for multiple tests.
+func ResetInstance(ctx context.Context, db *sql.DB) error {
```

Comment:
thought: Shouldn't this also drop databases? As is it feels like this is implemented specifically for how it's used in the AT suite.

Reply:
We actually have the AT's drop the database. This is more design cluster-scope values. I could totally see databases considered cluster-scope though... Right now, we will just keep the individual AT's dropping their own databases, since if they fail to do that, it indicates a bug.